doc/relevancy.md

```diff
@@ -0,0 +1,459 @@
+filter out when processing a search query, since they're so common
+they're not useful.  They also enable Elasticsearch to maintain a
+smaller index.
+
+The stopwords themselves are provided by Elasticsearch.  There's
+nothing in the Search API repo that looks at a list of omitted terms.
+This is handled by the default [stop token filter][].
+
+**History of stopwords**
+
+There has been some talk of not including stopwords in the past
+([0fe6e52][], May 2015) during this time a `no_stop` method was
+implemented that seems to stop stopwords from being used.  Again this
+seems to be served by Elasticsearch because it's not defined in Search
+API.
+
+Given that we're taking a stock list of stopwords from Elasticsearch,
+we may be missing out on a chance to edit them in a way that might be
+more useful for users.  I'm not sure how we'd go about changing this
+but I'd guess that we'd need to look into what words could be useful
+to omit / not omit.
+
+A search with stopwords is different to a search without stopwords:
+
+- If you search Search API for stopwords then you get literally
+  nothing: http://www.gov.uk/api/search.json?q=the+that+and+if
+
+- If you search for nothing then you get the some of the more
+  frequently clicked links: http://www.gov.uk/api/search.json?q=
+
+### Synonyms
+
+A synonym is a word or phrase that means exactly or nearly the same as
+another word or phrase in the same language, for example shut is a
+synonym of close.
+
+For example, we use synonyms to show relevant results related to
+"vehicle tax" regardless of whether you searched for "car tax" or
+"auto tax".
+
+Synonyms are defined in the [synonyms.yml][] file and are applied to
+the Elasticsearch index configuration in the [schema_config.rb][]
+file.
+
+#### Grouping of synonyms
+
+Each group of synonyms is written in the form:
+
+```
+foo, bar, baz => bat, qux, quux
+```
+
+This means that each of the (comma separated) terms on the left of the
+`=>` map to each of the (comma separated) terms on the right.
+
+For example:
+
+```
+leap, hop => jump
+```
+
+means "a search for 'leap' or 'hop' should return documents with
+'jump' in them.
+
+And:
+
+```
+run, speedwalk => sprint, gallop
+```
+
+means "a search for 'run' or 'speedwalk' should return documents with
+'sprint' or 'gallop' in them."
+
+It must be applied both at index time and at query time, to ensure that query
+terms are mapped to the same single value that exists in the index.
+
+### Filtering
+
+Additional configuration is defined in the [elasticsearch_schema.yml][] and
+[stems.yml][] files.  This configuration is merged with the JSON
+configuration, and then passed to Elasticsearch directly.
+
+[This blog post][synonyms-blog] suggests that as well as using
+keywords, it can be useful to use "keepwords" to only filter the
+phrases we want.
+
+The post suggests to first generate shingles, followed by synonyms,
+followed by keepwords.
```

Comment:
Would it help here to briefly explain some of these terms? Maybe like this, but with better explanations...

```
The post suggests to first generate shingles (a group every
possible 1 - N set of words as a single token), followed by
synonyms, followed by keepwords (a list of words to keep).
```